### PR TITLE
Don't stamp a stack footer onto single-PR stacks

### PR DESCRIPTION
**Pull request by krlvi**
## What this resolves

GitButler was appending `This is **part 1 of 1 in a stack** made with GitButler` to PR descriptions — including PRs the user never opened through GitButler. Reported in Discord: a user with the GitHub integration on saw the footer show up on a PR they'd created manually.

The exact situation that triggers it: a stack that has **2+ branches but only one open PR**. On a branch reorder (drag), the frontend rebuilds every PR body in that stack via `updateStackPrs` → `updateBody`. The guard in `updateStackPrs` checks the number of *branches* (`branchDetails.length <= 1`), not the number of *PRs*, so it proceeds; `allPrNumbers` then filters down to a single number, and the frontend `updateBody` — unlike its Rust counterpart `but_forge::update_body` — had no single-PR guard, so it generated a footer anyway: "part 1 of 1". Because `updateStackPrs` writes to any PR it can associate to a branch, this also rewrites PRs GitButler didn't create.

## Why it's showing up now

The frontend `updateBody` has always lacked the guard, but casual users couldn't easily reach the drag path that calls it. That changed with #14669 (*Support reordering empty branches in single-branch mode*) and #14713 (*Allow single-branch stack reordering*), which enabled branch-reorder drags in single-branch mode. In single-branch mode every branch shares one stack, so "2+ branches, one PR" is the normal state rather than an edge case — turning a latent bug into a common one.

## Fix

Mirror the single-PR guard from `but_forge::update_body` in the frontend: with one PR, add no footer and strip any existing one. The body head/footer/tail recomposition is unified into a single `composeBody` helper so `updateBody` and `clearFooter` can't drift on whitespace edge cases. Adds unit tests covering the single-PR, footer-removal, and multi-PR cases.

Follow-up (not in this PR): the whole frontend footer path duplicates `but_forge::sync_reviews`; the drag handlers should route through the Rust implementation so the two copies can't diverge again.

### CODE REVIEW

File: apps/desktop/src/lib/forge/shared/prFooter.ts [hunk 1]
```diff
@@ -121,32 +121,39 @@ export async function unstackPRs(
...
 	symbol: string,
 ) {
-	const head = (body?.split(STACKING_FOOTER_BOUNDARY_TOP).at(0) || "").trim();
-	const tail = (body?.split(STACKING_FOOTER_BOUNDARY_BOTTOM).at(1) || "").trim();
-	const footer = generateFooter(prNumber, allPrNumbers, symbol);
-	const description = head + "\n\n" + footer + "\n\n" + tail;
-	return description;
+	const footer =
+		allPrNumbers.length > 1 ? generateFooter(prNumber, allPrNumbers, symbol) : undefined;
+	return composeBody(body, footer);
```
[the remark this answers is not in the document]
Comment (krlvi) [reply apps/desktop/src/lib/forge/shared/prFooter.ts]: Good catch — done. The single-PR path now delegates to \`clearFooter\`, which guards on both boundary markers, so a footer-less body is returned untouched (no spurious rewrite) and a lone top marker no longer drops trailing content. Added tests for both cases.
